### PR TITLE
Deploy to production using GitHub Actions

### PR DESCRIPTION
**Pull request by Sybrand**
Using a workflow triggered by a deployment allows us to get rid of Jenkins, and aligns us better with introducing something like chatops.

Currently, to trigger deploy, you have to use the github api:

`gh api repos/bcgov/wps/deployments -f ref="[branch name]"`

### CODE REVIEW

File: .github/workflows/production.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        with:
+          # reference used to figure out this part:
+          # https://docs.github.com/en/rest/reference/pulls#list-pull-requests
+          # https://octokit.github.io/rest.js/v18#pulls
+          # https://github.com/actions/github-script
+          script: |
+            // get pull requests relating to this commit hash
+            const pulls = (await github.repos.listPullRequestsAssociatedWithCommit({owner: context.repo.owner, repo: context.repo.repo, commit_sha: context.sha }))
+            console.log(pulls)
+            // find an open PR (potentially problematic if you have multiple PR's open!)
```
Comment (conbrad) [inline .github/workflows/production.yml:25]: You can get this exact PR number from an environment variable, check out: https://github.com/bcgov/wps/blob/lnd/pr-github-action/.github/actions/pr-test-url/index.js

**Review by conbrad — CHANGES_REQUESTED** [review]
Nice

File: .github/workflows/production.yml (hunk 1 above)
Comment (Sybrand) [reply .github/workflows/production.yml:25]: It's not common to have multiple PR's at the same commit point. Thanks for the link - I'll test.

File: .github/workflows/production.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
+name: Production
+
+# trigger this deployment by running:
+# gh api repos/bcgov/wps/deployments -f ref="[your branch]"
+on:
+  # reference used to figure this part out:
+  # https://docs.github.com/en/actions/reference/events-that-trigger-workflows#deployment
+  deployment:
```
Comment (andrea-williams) [inline .github/workflows/production.yml:8]: is having the `:` after `deployment` correct?

**Review by andrea-williams — APPROVED** [review]

**Review by conbrad — APPROVED** [review]
🚀

File: .github/workflows/production.yml (hunk 2 above)
Comment (Sybrand) [reply .github/workflows/production.yml]: It doesn't break it - but sure - looks weird. I've removed it.

File: .github/workflows/production.yml (hunk 1 above)
Comment (Sybrand) [reply .github/workflows/production.yml]: @conbrad so actually that doesn't work - loading that file is the same as referencing context - and it doesn't have the PR in this context.

Comment (conbrad) [reply .github/workflows/production.yml]: Oops, sorry about that.

**Pull request merged**